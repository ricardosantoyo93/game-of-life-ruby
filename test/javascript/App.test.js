import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { App } from '../../app/javascript/packs/App';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
    let wrapper = null;

    beforeEach(() => {
        wrapper = shallow(<App run={false} />);
    });

    test('renders', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<App run={false} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should call saveGridToStore 4 times', () => {
        const setNewGrid = jest.fn();
        const instance = wrapper.instance();
        jest.spyOn(instance, "saveGridToStore").mockImplementation(setNewGrid);
        instance.rowsRef.current = {
            value: null
        }
        instance.colsRef.current = {
            value: null
        }
        wrapper.find('#createGrid1').simulate('click');
        wrapper.find('#createGrid2').simulate('click');
        wrapper.find('#createGrid3').simulate('click');
        wrapper.find('#randomGrid').simulate('click');
        expect(setNewGrid).toHaveBeenCalledTimes(4);
    });
});
