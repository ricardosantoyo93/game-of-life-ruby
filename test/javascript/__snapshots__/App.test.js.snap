// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
.c1 {
  background: #088;
  border-radius: 3px;
  border: 2px solid #088;
  color: white;
  margin: 0 1em;
  padding: 0.25em 1em;
  font-size: 16px;
  text-transform: uppercase;
}

.c1:disabled {
  background-color: gray;
}

.c2 {
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0 1em;
  padding: 0.25em 1em;
  font-size: 16px;
  text-transform: uppercase;
}

.c2:disabled {
  background-color: gray;
}

.c0 {
  background: transparent;
  border-radius: 3px;
  border: 2px solid #bbb;
  color: #bbb;
  margin: 0 1em;
  padding: 0.25em 1em;
  font-size: 16px;
  width: 100px;
  text-transform: uppercase;
  text-align: center;
}

.c0:disabled {
  background-color: gray;
}

<div
  className="App"
>
  <div
    className="App-container"
  >
    <p>
      Select grid dimensions
    </p>
    <br />
    <span>
      <input
        className="c0"
        disabled={false}
        max="50"
        min="1"
        placeholder="Rows"
        type="number"
      />
      x
      <input
        className="c0"
        disabled={false}
        max="50"
        min="1"
        placeholder="Columns"
        type="number"
      />
      <button
        className="c1"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Create Grid
      </button>
    </span>
    <br />
    <span>
      <button
        className="c2"
        disabled={false}
        id="createGrid1"
        onClick={[Function]}
        type="button"
      >
        20 x 30
      </button>
      <button
        className="c2"
        disabled={false}
        id="createGrid2"
        onClick={[Function]}
        type="button"
      >
        30 x 40
      </button>
      <button
        className="c2"
        disabled={false}
        id="createGrid3"
        onClick={[Function]}
        type="button"
      >
        40 x 50
      </button>
      <button
        className="c2"
        disabled={false}
        id="randomGrid"
        onClick={[Function]}
        type="button"
      >
        Random
      </button>
    </span>
  </div>
</div>
`;
